bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/transform/math/internal/AbstractMathTransformationService.java

```diff
@@ -41,22 +39,19 @@ abstract class AbstractMathTransformationService implements TransformationServic
         }
 
         QuantityType<?> source;
-        try {
-            source = new QuantityType<>(sourceString);
-        } catch (IllegalArgumentException e) {
-            logger.warn("Input value '{}' could not be converted to a valid number", sourceString);
-            throw new TransformationException("Math Transformation can only be used with numeric inputs");
-        }
         QuantityType<?> value;
+        String evaluated = sourceString;
         try {
-            value = new QuantityType<>(valueString);
+            source = new QuantityType<>(evaluated);
+            evaluated = valueString;
+            value = new QuantityType<>(evaluated);
         } catch (IllegalArgumentException e) {
-            logger.warn("Input value '{}' could not be converted to a valid number", valueString);
+            logger.warn("Input value '{}' could not be converted to a valid number", evaluated);
             throw new TransformationException("Math Transformation can only be used with numeric inputs");
         }
+        
         try {
-            QuantityType<?> result = performCalculation(source, value);
-            return BigDecimal.ZERO.compareTo(result.toBigDecimal()) == 0 ? "0" : result.toString();
+            return performCalculation(source, value).toString();
```

Comment:
I See why it works now, but I wonder why it was done like that before. Maybe to prevent a return value of `0.0000`?

Reply:
Yes, I was wondering the same, reason why I wanted your advice.